### PR TITLE
Task Query - Error Scenarios

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-592

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,1431 @@
...
+            ).build();
+
+        return Stream.of(
+            publicClassification,
+            privateClassification,
+            restrictedClassification
+        );
+    }
+
+    private static Stream<TaskQueryScenario> grantTypeBasicSpecificErrorScenario() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:349]: looks like typo. Should the name be grantTypeSpecificErrorScenario

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,1431 @@
...
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+        final Map<String, String> tcAttributes = Map.of(
+            RoleAttributeDefinition.CASE_TYPE.value(), "Asylum",
+            RoleAttributeDefinition.JURISDICTION.value(), "IA",
+            RoleAttributeDefinition.CASE_ID.value(), "1623278362431003"
+        );
+        RoleAssignment roleAssignment = RoleAssignment.builder().roleName("tribunal-caseworker")
+            .classification(Classification.RESTRICTED)
+            .beginTime(LocalDateTime.now().minusYears(1))
+            .authorisations(List.of("DIVORCE", "PROBATE"))
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:360]: do we also need too capture scenario when authorisation is null or empty?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,1431 @@
...
+        final Map<String, String> tcAttributes = Map.of(
+            RoleAttributeDefinition.REGION.value(), "1",
+            RoleAttributeDefinition.JURISDICTION.value(), "IA",
+            RoleAttributeDefinition.BASE_LOCATION.value(), "765324"
+        );
+        RoleAssignment roleAssignment = RoleAssignment.builder().roleName("standard-caseworker")
+            .classification(Classification.RESTRICTED)
+            .attributes(tcAttributes)
+            .grantType(GrantType.STANDARD)
+            .authorisations(List.of("unknownValue"))
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:493]: do we also need to capture scenario when authorisation is null or empty?

**Review by BenNzewi — COMMENTED** [review]
Should we remove AC5? In the context of the query, we cant return all grantTypes so looks invalid.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java (hunk 2 above)
Comment (indvamshi) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:360]: Its already covered in HappyPath

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java (hunk 3 above)
Comment (indvamshi) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:493]: covered in happy path.

**Pull request closed without merging**